commons/commons-util/src/main/kotlin/ebi/ac/uk/io/FileUtils.kt

```diff
@@ -1,88 +1,151 @@
 package ebi.ac.uk.io
 
+import ebi.ac.uk.io.FileUtilsHelper.createFileHardLink
+import ebi.ac.uk.io.FileUtilsHelper.createFolderHardLinks
+import ebi.ac.uk.io.FileUtilsHelper.createFolderIfNotExist
+import ebi.ac.uk.io.FileUtilsHelper.createParentDirectories
 import java.io.File
 import java.nio.file.Files
+import java.nio.file.Files.exists
 import java.nio.file.Path
 import java.nio.file.StandardCopyOption
+import java.nio.file.attribute.FileAttribute
+import java.nio.file.attribute.PosixFilePermission
+import java.nio.file.attribute.PosixFilePermissions
+import java.nio.file.attribute.PosixFilePermissions.asFileAttribute
 import kotlin.streams.toList
 
+val ONLY_USER: Set<PosixFilePermission> = PosixFilePermissions.fromString("rwx------")
+
 @Suppress("TooManyFunctions")
 object FileUtils {
-    fun deleteFolder(file: File) {
-        deleteFolder(file.toPath())
+    fun copyOrReplaceFile(
+        source: File,
+        target: File,
+        permissions: Set<PosixFilePermission>
+    ) {
+        when (isDirectory(source)) {
+            true -> FileUtilsHelper.copyFolder(source.toPath(), target.toPath(), asFileAttribute(permissions))
+            false -> FileUtilsHelper.copyFile(source.toPath(), target.toPath(), asFileAttribute(permissions))
+        }
+    }
+
+    fun getOrCreateFolder(
+        folder: Path,
+        permissions: Set<PosixFilePermission>
+    ): Path {
+        require(exists(folder).not() || isDirectory(folder.toFile())) { "'$folder' points to a file" }
+        createFolderIfNotExist(folder.parent, permissions)
+        createFolderIfNotExist(folder, permissions)
+        return folder
     }
 
-    fun reCreateDirectory(file: File): File {
-        val path = file.toPath()
-        deleteFolder(path)
-        Files.createDirectory(path)
+    fun reCreateFolder(file: File, permissions: Set<PosixFilePermission>): File {
```

Comment:
This is just a suggestion but, if the method is for folders I think it might be a bit confusing that the parameter is named "file"